lib/releases.sh

```diff
@@ -1,44 +1,27 @@
 #!/bin/bash
 
 function get_latest_release() {
-  # get last page
+  
+  # fail when release_path is not passed
+  local release_path="${1:?no release path is given}"
+  
+  # set url to get all releases from github
+  local url="${release_path}?per_page=100"
+
+  # fail when release is not passed
+  local release="${2:?no release given}"
+  
+  # This gets the latest release as vx.y.z or vx.y.z-rc.0, including any version with a suffix starting with - , for example -rc.0
+  # The order is exactly as released in Github.
+  # Downside is that selecting official releases only isn't possible, while pre-release
+  # selection is possible given specific enough prefix, like v1.3.0-pre
+
   if [ -z "${GITHUB_TOKEN:-}" ]; then
-    last_page="$(curl -s -I "${1}" | grep '^link:' | sed -e 's/^link:.*page=//g' -e 's/>.*$//g')"
+    release_tag=$(curl -sL "${url}" | jq .[].name -r | grep -E "${release}" -m 1)
```

Comment:
Sorry for nitpicking this much, but please put `.[].name` in quotes for safer handling, and add the missing `^` in front of `${release}` so it matches releases from the start only as it should.

```suggestion
    release_tag=$(curl -sL "${url}" | jq ".[].name" -r | grep -E "^${release}" -m 1)
```